packages/pigeon/lib/ast.dart

```diff
@@ -82,39 +74,72 @@ class Api extends Node {
   }
 }
 
-/// Represents a field on a [Class].
-class Field extends Node {
-  /// Parametric constructor for [Field].
-  Field({
-    required this.name,
-    required this.dataType,
+/// An entity that represents a typed concept, like a [TypeDeclaration] or [NamedType].
+abstract class TypedEntity {
+  /// The datatype base name of the entity (ex 'Foo' to 'Foo<Bar>?').
+  String get baseName;
```

Comment:
Yea, I see your point.  I renamed baseName to typeBaseName at the TypedEntity level.  Now it reads more properly in the TypeDeclaration and reads well for the TypedEntity too, matching what we did with typeArguments.

Reply:
Talked offline a bit about this.  I think I have a better understanding of @stuartmorgan's point and why this isn't resolved yet.  There are definitely cases where having an adapter to rename something from an interface is a good idea.  I don't think that I've run into them in the wild much which made it hard to grok.  My hope was that this renaming would make the adapter unnecessary if we chose the right name in the interface.  In my eye it foot the bill, but stuart has fresh eyes so if it doesn't jive with him it's worth reconsidering.  The problem with the adapter is that if we can't find a name that works in both places, unfortunately the adapter doesn't hide the name we thought was confusing in the first place ie if `TypeDeclaration.baseName` is confusing, having `TypeDeclaration.baseName` and `TypeDeclaration.typeBaseName` is more confusing imo.

So, if we can't figure out a name that we are happy with in both contexts, it's preferable to eliminate TypedEntity.  The fact that we moved from Field to NamedType makes TypedEntity even less useful and may be more of a relic of the mental model I was using for the code before other changes we made in this PR.

What do you think @stuartmorgan, does that sound good, yank TypedEntity?

Reply:
> unfortunately the adapter doesn't hide the name we thought was confusing in the first place ie if `TypeDeclaration.baseName` is confusing, having `TypeDeclaration.baseName` and `TypeDeclaration.typeBaseName` is more confusing imo.

`TypeDeclaration.baseName` isn't confusing, it's the right name. `TypeDecalaration.typeBaseName` is just weirdly verbose as a primary API because it pointlessly repeats "type" (just as per offline discussion, the name field for a `Person` class should be `name`, not `personName`).

So the problem isn't that one of the APIs is confusing, but that the other one—the one it needs to implement to conform to `TypedEntity`—is cumbersome in the context of the class (but not when accessed through `TypedEntity`). We agree that having both is still not great, but it's not nearly as bad as the reverse (where `TypedEntity` had `baseName`, and thus there was a `Field.baseName` whose meaning was actually totally unclear).

> So, if we can't figure out a name that we are happy with in both contexts

I do not believe that in this case there can be a name that fits both contexts. In `TypedEntity` the name *must* indicate that it's related to the type, since that's not at all evident in generic implementors (such as `Field`), but indicating that it's related to a type will always be redundant in the context of `TypeDeclaration` as that's a data class whose sole purpose is describing a type.

> What do you think @stuartmorgan, does that sound good, yank TypedEntity?

I'm fine with removing `TypedEntity`.